### PR TITLE
Cuckoo API integration

### PR DESCRIPTION
**Pull request by Jack28**
I have added support to connect to CuckooSandbox via its API

### CODE REVIEW

File: peekaboo/toolbox/cuckoo.py [hunk 1]
```diff
@@ -172,37 +355,52 @@ class CuckooReport(object):
...
-                % (self.job_id, cuckoo_report)
-            )
-            self.file_path = cuckoo_report
-            with open(cuckoo_report) as data:
-                try:
-                    report = json.load(data)
-                    self.report = report
-                except ValueError as e:
-                    logger.exception(e)
+            raise Exception("Invalid report source given")
```
Comment (SebastianDeiss) [inline peekaboo/toolbox/cuckoo.py]: Should be ``ValueError``.

File: peekaboo/config.py [hunk 2]
```diff
@@ -89,6 +89,10 @@ def __init__(self, config_file='./peekaboo.conf'):
         self.keep_mail_data = None
         self.db_url = None
         self.ruleset_config = None
+        self.cuckoo_mode = "api"
+        self.cuckoo_dummy_report = None
+        self.cuckoo_url = "http://127.0.0.1:8090"
```
Comment (SebastianDeiss) [inline peekaboo/config.py]: You should not set actual config values here.

File: peekaboo/toolbox/cuckoo.py [hunk 3]
```diff
@@ -40,47 +43,227 @@
...
+    def do(self):
+        # reaktor and shit
+        # Run Cuckoo sandbox, parse log output, and report back of Peekaboo.
+        srv = CuckooServer()
+        reactor.spawnProcess(srv, self.interpreter, [self.interpreter, '-u',
+                                                     self.cuckoo_exec])
+        reactor.run()
+
+
+class CuckooDummy(Cuckoo):
```
Comment (SebastianDeiss) [inline peekaboo/toolbox/cuckoo.py]: ``CuckooDummy`` is for development purposes only. So, it should be a part of the unit tests or a separate script.

**Comment by Jack28** [comment]
Have a look again please

**Comment by SebastianDeiss** [comment]
LGTM. Do you want to add another dummy Cuckoo with this pull or open a new one for that.

File: peekaboo/daemon.py [hunk 4]
```diff
@@ -211,13 +210,17 @@ def run():
...
-        # Run Cuckoo sandbox, parse log output, and report back of Peekaboo.
-        # If this dies Peekaboo dies, since this is the main thread.
-        srv = CuckooServer()
-        reactor.spawnProcess(srv, config.interpreter, [config.interpreter, '-u',
-                                                       config.cuckoo_exec])
+        # If this dies Peekaboo dies, since this is the main thread. (legacy)
+        if config.cuckoo_mode == "embed":
+            cuckoo = CuckooEmbed(config.interpreter, config.cuckoo_exec)
+        elif config.cuckoo_mode == "dummy":
+            cuckoo = CuckooDummy()
```
Comment (SebastianDeiss) [inline peekaboo/daemon.py]: @Jack28 you forgot to remove ``CuckooDummy`` from ``daemon.py``.

Comment (Jack28) [reply peekaboo/daemon.py]: I've noticed that. There is a commit on my fork/branch that fixes this. How do you want me to include this?
Revert and merge again, commit to master ... ?

**Pull request merged**